testing/sdk_auth.py

```diff
@@ -283,11 +279,14 @@ def add_principals(self, principals: list):
         self.principals = principals
 
         log.info("Adding the following list of principals to the KDC: {principals}".format(principals=principals))
-        add_principal_cmd = "task exec {task_id} /usr/sbin/kadmin -l add --use-defaults --random-password {principal}"
+        kadmin_options = ["-l"]
+        kadmin_cmd = "add"
+        kadmin_args = ["--use-defaults", "--random-password"]
+
         for principal in principals:
             try:
-                sdk_cmd.run_cli(add_principal_cmd.format(task_id=KERBEROS_APP_ID, principal=principal))
-            except dcos.errors.DCOSException as e:
+                self.__run_kadmin(kadmin_options, kadmin_cmd, kadmin_args + [principal])
```

Comment:
As far as I know, `kadmin` supports adding the principals in a list. So one could use:
```bash
kadmin_args.extend(principals)
self.__run_kadmin(kadmin_options, kadmin_cmd, kadmin_args)
```
I'm not sure how it handles, for example, existing principals in this case.

Reply:
Yeah, all principals can be added simultaneously.